src/cli/shell_hook.rs

```diff
@@ -0,0 +1,70 @@
+use std::path::PathBuf;
+
+use miette::IntoDiagnostic;
+use rattler_conda_types::Platform;
+use rattler_shell::{
+    activation::{ActivationVariables, Activator},
+    shell::ShellEnum,
+};
+
+use crate::prefix::Prefix;
+
+/// Generates the activation script.
+fn generate_activation_script() -> miette::Result<String> {
+    let prefix = Prefix::new(PathBuf::new())?;
+    let shell: ShellEnum = if cfg!(windows) {
```

Comment:
I think we should take this as option via args, e.g. `--shell bash/zsh/cmd.exe/powershell/pwsh/fish/...` 

And only use bash & cmd.exe as defaults.

Reply:
And I thinknthe defaults are also define in rattler_shell if Im not mistaken

Reply:
Added the option in d21d383f81a377aee79f013ab188683ad2b61dff